src/main/java/Duke.java

```diff
@@ -1,10 +1,212 @@
+                    tasks.add(newEvent);
+
+
+                    System.out.println("____________________________________________________________");
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + newEvent);
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                    System.out.println("____________________________________________________________");
+                } else if (userInput.startsWith("delete")) {
+                    int taskNumber;
+                    try {
+                        taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+                        if (taskNumber <= 0 || taskNumber > tasks.size()) {
+                            throw new DukeException("Invalid task number!");
+                        }
+
+                        Task deletedTask = tasks.get(taskNumber - 1);
+
+                        tasks.remove(taskNumber - 1);
+
+                        System.out.println("____________________________________________________________");
+                        System.out.println("Noted. I've removed this task: ");
+                        System.out.println("  " + deletedTask);
+                        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                        System.out.println("____________________________________________________________");
+                    } catch (NumberFormatException e) {
+                        System.out.println("____________________________________________________________");
+                        System.out.println("Please provide a valid task number.");
+                        System.out.println("____________________________________________________________");
+                    } catch (DukeException de) {
+                        System.out.println("____________________________________________________________");
+                        System.out.println(de.getMessage());
+                        System.out.println("____________________________________________________________");
+                    }
+                } else {
+                    throw new DukeException("☹ OOPS!!! What is that? I don't know what is that? What is the meaning :-(");
+                }
+            } catch (DukeException de) {
+                System.out.println("____________________________________________________________");
+                System.out.println(de.getMessage());
+                System.out.println("____________________________________________________________");
+            }
+            saveTasksToFile();
+        }
+
+        System.out.println("____________________________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
```

Comment:
1. This is a particularly long method. There are numerous opportunities to refactor it. Please try to reduce the code by refactoring some repeated elements.
2. While refactoring, try to remove the arrowhead nature of the code.